### ISSUE TITLE
InputData source not (yet?) support for this format (Smile, CBOR)

### ISSUE THREAD

**Issue by henrik-lindqvist** [FasterXML/jackson-dataformats-binary#29]
Seems the createParser method of SmileFactory and CBORFactory is lacking implementation for java.io.DataInput sources.

**Comment by cowtowncoder** [comment]
Yes, this is correct. Due to time it took to add implementation for json backend, I had to postpone implementation for CBOR and Smile which would be the next likely cases (protobuf, alas, may be not supportable without external framing).

Would you be interested in using such input source if it was available? And if so, any preference order?

Note, too, that creating backend should be quite simple and mechanical.
However, I don't think it is doable by refactoring just because byte access is on critical path: that is, addition of `nextByte()` (or whatever) abstraction would be performance problem. This based on json implementation experiences.

**Comment by henrik-lindqvist** [comment]
No great importance. Glancing at the source, with little in-depth knowledge, i though it was a minor job, overriding a method. Simple workaround, using a DataInput to InputStream wrapper for now.

**Comment by cowtowncoder** [comment]
Ok. Mostly asking since I hope to gauge interest in yet-to-add/complete features, and if there's demand for `InputData` implementation I would consider it for implementation. There are so many things to potentially work on that I prefer working on things that seem more likely to be used and useful. :)

**Comment by henrik-lindqvist** [comment]
Focus on the more useful things! As said there's a simple workaround anyway.

**Comment by cowtowncoder** [comment]
@henrik-lindqvist  :)

**Comment by leonard84** [comment]
Are there any plans to implement this? I just stumbled over this today.

**Comment by cowtowncoder** [comment]
@leonard84 I don't think anyone is working on this. Usually what happens is that people add a note on issue if they work on it.
